c/dataservice.c

```diff
@@ -214,20 +214,40 @@ WebPlugin *makeWebPlugin(char *pluginLocation, JsonObject *pluginDefintion, Inte
     for (int i = 0; i < jsonArrayGetCount(dataServices); i ++) {
       JsonObject *serviceDef = jsonArrayGetObject(dataServices, i);
       char *type = jsonObjectGetString(serviceDef, "type");
-
-      if (!type || !strcmp(type, "service")) {
+      char *serviceName = jsonObjectGetString(serviceDef, "name");
+      char *sourceName = jsonObjectGetString(serviceDef, "sourceName");
+      bool isImport = strcmp(type, "import") ? false : true;
+      if (!type) {
+        printf("*** PANIC: Returning NULL for plugin. Check pluginDefinition for correct 'type' fields on dataservices. ***\n");
+        plugin = NULL;
+        safeFree((char*)plugin,sizeof(WebPlugin));
+        return NULL;
+      } else if (!isImport && !serviceName) {
```

Comment:
It seems like in the first 3 ```if``` statements you're trying to validate the input and they could be isolated in their own validate function, then you'd have a simple code like this:
```c
...
/* validate */
if (!isServiceDefinionValid(serviceDef)) {
  safeFree((char *)plugin, sizeof(WebPlugin));
  plugin = NULL;
  return NULL;
}

/* determine the type */
...
````